packages/realm-web/src/auth-providers/ApiKeyAuthProvider.test.ts

```diff
@@ -0,0 +1,190 @@
+////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2020 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////
+
+import { expect } from "chai";
+import { ObjectId } from "bson";
+
+import { ApiKeyAuthProvider } from "./ApiKeyAuthProvider";
+import { MockTransport } from "../test/MockTransport";
+
+const DEFAULT_HEADERS = {
+    Accept: "application/json",
+    "Content-Type": "application/json",
+};
+
+describe("ApiKeyAuthProvider", () => {
+    it("can create an api key", async () => {
+        const transport = new MockTransport([
+            {
+                _id: {
+                    $oid: "deadbeefdeadbeefdeadbeef",
+                },
+                name: "my-key-name",
+                key: "super-secret-key",
```

Comment:
Should we call it `value` instead of `key`?

Reply:
This is testing the interface towards the server, so that can't be changed without a server-side update.